google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -171,6 +185,11 @@ private static String getValue(Label label) {
 
   /* Detect monitored Resource type using environment variables, else return global as default. */
   private static Resource getAutoDetectedResourceType() {
+    if (System.getenv("K_SERVICE") != null &&
```

Comment:
Steren's note: Note that checking the presence of K_SERVICE and other env vars is not sufficient to differentiate between Cloud Run and any other Knative installation (e.g. Cloud Run for Anthos).

Would adding a check for "KUBERNETES_SERVICE_HOST" == null, fix this issue?

Reply:
Yes, that makes sense.